### ISSUE TITLE
Delete cached attachments

### ISSUE THREAD

**Issue by jmcarp** [eregs/notice-and-comment#70]
Attachments are uploaded from the client to S3 and then sent to regs.gov on comment submit. We don't currently delete attachments from S3 after we're done with them. We could delete attachments immediately after successful comment submission--but if we want to email submitted comments to users with links to attachments, we'll need to keep them around longer. Implementation options include:

* Use S3 object lifecycle management to delete attachments after a fixed interval: http://docs.aws.amazon.com/AmazonS3/latest/dev/object-lifecycle-mgmt.html
* Use a celery chain or success callback to delete attachments after submission

cc @cmc333333 @vrajmohan

**Comment by jmcarp** [comment]
I can implement this once I understand the desired behavior. At the moment, file preview links last for 90 days, so a simple approach would be to expire files 90 days after upload. Ping @cmc333333 @vrajmohan.

**Comment by cmc333333** [comment]
That sounds fine; we can tweak it if desired later.

**Comment by jmcarp** [comment]
This turns out not to be allowed with the current cloud.gov service. We can add a celery callback to delete attachments after a successful submit, although that's not going to clean up files associated with pdf previews and failed submits. We could also use prefixes to break up cached files by doc id, then delete all files associated with a doc id and delete them after the comment period expires. Then we could add a nightly / weekly task to celery-cron and clear out all prefixes for proposals whose comment periods have expired.

What do you think @cmc333333 @vrajmohan?

**Comment by vrajmohan** [comment]
Is this the situation whether we have S3 as attached service (as we currently do) or separate S3 bucket (https://github.com/18F/Infrastructure/issues/589)?

**Comment by jmcarp** [comment]
@vrajmohan: good point--if we use a bucket that's not managed by cloud.gov, we should be able to set lifecycle rules. But my impression is that infrastructure prefers that we stick to services managed by cloud.gov. We can ask about that and see if they'd be willing to make an exception here, if we think it's worth pushing.
